web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -167,55 +173,220 @@ file systems will be created on the selected device.");
+  const [editedDevices, setEditedDevices] = useState(false);
+
+  const selectBootDevice = () => {
+    setIsBootDeviceSelected(true);
+    onValidate(true);
+  };
+
+  const selectCustomDevices = () => {
+    setIsBootDeviceSelected(false);
+    const { bootDevice } = settings;
+    const customDevices = (vgDevices.length === 0 && !editedDevices) ? [bootDevice] : vgDevices;
+    setVgDevices(customDevices);
+    onValidate(customDevices.length > 0);
+  };
+
+  const onChangeDevices = (devices) => {
+    setVgDevices(devices);
+    setEditedDevices(true);
+    onValidate(devices.length > 0);
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    const customDevices = isBootDeviceSelected ? [] : vgDevices;
+    onSubmitProp(customDevices);
+  };
+
+  const BootDevice = () => {
+    const bootDevice = devices.find(d => d.name === settings.bootDevice);
+
+    return <DeviceList devices={[bootDevice]} />;
+  };
+
+  return (
+    <Form id={id} onSubmit={onSubmit}>
+      <div className="split">
+        <span>{_("Devices for creating the volume group")}</span>
+        <ToggleGroup aria-label="Volume group devices" isCompact>
+          <ToggleGroupItem
+            text={_("Installation device")}
+            buttonId="bootDevice"
+            isSelected={isBootDeviceSelected}
+            onClick={selectBootDevice}
+          />
+          <ToggleGroupItem
+            text={_("Custom devices")}
+            buttonId="customDevices"
+            isSelected={!isBootDeviceSelected}
+            onClick={selectCustomDevices}
+          />
+        </ToggleGroup>
+      </div>
+      <If
+        condition={isBootDeviceSelected}
+        then={<BootDevice />}
+        else={
+          <DeviceSelector
+            isMultiple
+            selected={vgDevices}
+            devices={devices}
+            onChange={onChangeDevices}
+          />
+        }
+      />
+    </Form>
+  );
+};
+
+/**
+ * Allows to select LVM and configure the system volume group.
  * @component
  *
  * @param {object} props
- * @param {boolean} props.selected - Whether LVM is selected
- * @param {boolean} props.isLoading - Whether to show the selector as loading
- * @param {onChangeFn} props.onChange - On change callback
+ * @param {ProposalSettings} props.settings - Settings used for calculating a proposal.
+ * @param {StorageDevice[]} [props.devices=[]] - Available storage devices.
+ * @param {boolean} [props.isChecked=false] - Whether LVM is selected.
+ * @param {boolean} [props.isLoading=false] - Whether to show the selector as loading.
+ * @param {onChangeFn} [props.onChange=noop] - On change callback.
  *
  * @callback onChangeFn
  * @param {boolean} lvm
  */
-const LVMField = ({ selected: selectedProp, isLoading, onChange }) => {
-  const [selected, setSelected] = useState(selectedProp);
+const LVMField = ({
+  settings,
+  devices = [],
+  isChecked: isCheckedProp = false,
+  isLoading = false,
+  onChange: onChangeProp = noop
```

Comment:
NP: Same than above here. I'd use `handleOnChange` internally instead of renaming the prop here.

However I have no idea now to avoid the `isChecked` rename.